contrib/win32/openssh/OpenSSHTestHelper.psm1

```diff
@@ -158,24 +158,20 @@ WARNING: Following changes will be made to OpenSSH configuration
     # copy new sshd_config
     Copy-Item (Join-Path $Script:E2ETestDirectory sshd_config) (Join-Path $script:OpenSSHBinPath sshd_config) -Force
     
+    Start-Service ssh-agent
+
     #copy sshtest keys
     Copy-Item "$($Script:E2ETestDirectory)\sshtest*hostkey*" $script:OpenSSHBinPath -Force    
     Get-ChildItem "$($script:OpenSSHBinPath)\sshtest*hostkey*"| % {
         #workaround for the cariggage new line added by git before copy them
         (Get-Content $_.FullName -Raw).Replace("`r`n","`n") | Set-Content $_.FullName -Force
         Adjust-HostKeyFileACL -FilePath $_.FullName
         if (-not ($_.Name.EndsWith(".pub"))) {
-            Add-PermissionToFileACL -FilePath $_.FullName -User "NT Service\sshd" -Perm "Read"
+            #register private key with agent
+            ssh-add-hostkey.ps1 $_.FullName
```

Comment:
This script does not have any dependencies. It can be called from anywhere. Already tested it.